README.md

```diff
@@ -99,94 +89,98 @@ The tests directory (augur-core/tests) contain tests and test fixtures to test t
-augur-contracts % npm version [<version>, major,minor,patch, prerelease]
-augur-contracts % git push && git push --tags
-# Publish th NPM, if this is a pre-release tag, deploy it to the dev channel. This dev channel is the default for version published from CI.
-augur-contracts % npm publish [--tag dev]
-```
+### General information about Augur
 
-## Additional notes
+- [A Roadmap For Augur and What's Next](https://medium.com/@AugurProject/a-roadmap-for-augur-and-whats-next-930fe6c7f75a)
+- [Augur Master Plan](https://medium.com/@AugurProject/augur-master-plan-42dda65a3e3d)
+
+### Terminology
+
+[Augur Terminology](http://blog.augur.net/faq/all-terms/)
+
+### Security model
+
+The security model is based on the idea that reporters will want to invest in a market where honesty prevails, and will not trust fraudulent markets. For example, assume that there exist two almost identical universes: an honest universe, where the factually correct answer is the predominant answer, and a fraudulent universe, where incorrect answers are predominant in order to cheat the system. The REP in the fraudulent universe will approach 0, and the REP in the honest universe will approach the theoretical yield. This phenomenon occurs because reporters do not want to trade on markets in a dishonest universe. Therefore, they will not create markets in a dishonest universe, and the REP holders will not receive any market fees.
+
+In order to create a fraudulent universe, an attacker must fork the main universe. They will need to buy a large share of REP and stake it on the lie outcome, as nobody else will want to engage in a deceitful market. In principle, the attacker will lose up to 2.5 times the maximum profit made from a successful attack, and their REP will approach 0.
+
+The reentrancy guard (libraries/ReentrancyGuard.sol) and controller (Controller.sol) contracts also help uphold security. One can also attack Ethereum contracts by creating a chain of wanted contract calls through a malicious user. The reentrancy guard protects against that by preventing someone from re-entering the contracts before they return. The controller contract serves to update Augur and stores a registry of whitelisted contract addresses. The controller also can perform administrative actions, such as transferring ownership and performing emergency stop functions. It can be operated in either the developer mode, where the developer updates contracts, or the decentralized mode, when people must upload contracts to make changes and therefore, the market creators/traders must cancel orders and shift to the new contract. For bugs found involving money or REP, contracts are locked / an escape hatch is enabled which allows withdrawal of shares (and cancelling orders) for a value half-way between the bid-ask spread at the time of the lock. In addition, another escape hatch is enabled, which allows withdrawal of any locked up REP.
```

Comment:
The mention of ReentrancyGuard, Controller, and Escape Hatches seem out of place here.  Consider just including these as doc-comments on the respective contracts.  I'm generally a fan of non-obvious classes (contracts) having a doc comment explaining their purpose (but not their implementation).  I think having text that describes the purpose of source code in the readme just leads to it going out of date much quicker, and also makes it non-obvious for the reader on where to find the information (it is disconnected).